toolchain/lowering/lowering_handle.cpp

```diff
@@ -24,16 +24,39 @@ auto LoweringHandleAddressOf(LoweringFunctionContext& context,
   context.SetLocal(node_id, context.GetLocal(node.GetAsAddressOf()));
 }
 
-auto LoweringHandleArrayType(LoweringFunctionContext& /*context*/,
-                             SemanticsNodeId /*node_id*/, SemanticsNode node)
+auto LoweringHandleArrayValue(LoweringFunctionContext& context,
+                              SemanticsNodeId node_id, SemanticsNode node)
     -> void {
-  CARBON_FATAL() << "TODO: Add support: " << node;
-}
+  auto* llvm_type = context.GetType(node.type_id());
+  auto* alloca =
+      context.builder().CreateAlloca(llvm_type, /*ArraySize=*/nullptr, "array");
+  context.SetLocal(node_id, alloca);
+  auto array_size = llvm_type->getArrayNumElements();
+  auto tuple_node_id = node.GetAsArrayValue();
+  auto* tuple_value = context.GetLocal(tuple_node_id);
+  auto* tuple_type =
+      context.GetType(context.semantics_ir().GetNode(tuple_node_id).type_id());
 
-auto LoweringHandleArrayValue(LoweringFunctionContext& /*context*/,
-                              SemanticsNodeId /*node_id*/, SemanticsNode node)
-    -> void {
-  CARBON_FATAL() << "TODO: Add support: " << node;
+  // For return values or call instructions, the tuple is not in the memory so
+  // it cannot be accessed directly to create the array value. Hence, we are
+  // creating a store in this block.
```

Comment:
Thanks for explaining this, but I'm still uncertain about this issue. If the tuple weren't in memory, then I would've thought to see a load. Do you have a reference for how you reached this conclusion and solution?

I was looking at the similar C++ code, and I see `extractvalue` gets used: https://cpp.compiler-explorer.com/z/Wq31ajrv4 -- maybe we should investigate that? A TODO would be fine:

```
// TODO: Investigate if `extractvalue` can be used to avoid copying non-pointer data.
```

Reply:
I discussed this with @zygoloid before implementing this.

Reply:
Thank you for the suggestion! Updated using `extractvalue`.